classes/mail/transport/PHPMailTransport.php

```diff
@@ -0,0 +1,170 @@
+            throw new Exception('Can\'t send raw message with phpmail');
+        }
+
+        $phpmailerMessage = $this->getPHPMailMessage($symfonyMessage);
+        try {
+            $success = $phpmailerMessage->Send();
+            if (!$success) {
+                error_log($phpmailerMessage->ErrorInfo);
+                return null;
+            }
+        } catch (Exception $e) {
+            error_log($phpmailerMessage->ErrorInfo);
+            return null;
+        }
+
+        return new SentMessage($symfonyMessage, $envelope ?? Envelope::create($symfonyMessage));
+    }
+
+    public function __toString(): string
+    {
+        return 'phpmail';
+    }
+
+    /**
+     * We won't send symfony message; transfer data to PHPMailer to send with mail()
+     */
+    protected function getPHPMailMessage(Email $symfonyMessage): PHPMailer
+    {
+        $mailer = new PHPMailer();
+        $mailer->isMail();
+        $mailer->IsHTML();
+        $mailer->Encoding = PHPMailer::ENCODING_BASE64;
+        $mailer->CharSet = $symfonyMessage->getHtmlCharset() ?? 'utf-8';
+        $mailer->XMailer = 'Public Knowledge Project Suite v3';
+        $mailer->WordWrap = static::MAIL_WRAP;
+
+        $request = Application::get()->getRequest();
+        $fromAddresses = $symfonyMessage->getFrom();
+        if (!empty($fromAddresses)) {
+            $f = current($fromAddresses);
+            if (Config::getVar('email', 'force_default_envelope_sender') && Config::getVar('email', 'default_envelope_sender') && Config::getVar('email', 'force_dmarc_compliant_from')) {
+                /* If a DMARC compliant RFC5322.From was requested we need to promote the original RFC5322.From into a Reply-to header
+                 * and then munge the RFC5322.From */
+                $alreadyExists = false;
+                foreach ($symfonyMessage->getReplyTo() as $r) {
+                    if ($r->getAddress() === $f->getAddress()) {
+                        $alreadyExists = true;
+                    }
+                }
+                if (!$alreadyExists) {
+                    $mailer->AddReplyTo($f->getAddress(), $f->getName());
+                }
+
+                $site = $request->getSite();
+
+                // Munge the RFC5322.From
+                if (Config::getVar('email', 'dmarc_compliant_from_displayname')) {
+                    $patterns = ['#%n#', '#%s#'];
+                    $replacements = [$f->getName(), $site->getLocalizedTitle()];
+                    $fromName = preg_replace($patterns, $replacements, Config::getVar('email', 'dmarc_compliant_from_displayname'));
+                } else {
+                    $fromName = '';
+                }
+                $fromEmail = Config::getVar('email', 'default_envelope_sender');
+            }
+            // this sets both the envelope sender (RFC5321.MailFrom) and the From: header (RFC5322.From)
+            $mailer->SetFrom($fromEmail ?? $f->getAddress(), $fromName ?? $f->getName());
```

Comment:
As `$fromName` might be an empty string, maybe here it should be `$fromName ?: $f->getName()`?

Reply:
Missed that, thanks!

Reply:
Great, so just a comment/test could be added to prevent people like me from breaking things 😂